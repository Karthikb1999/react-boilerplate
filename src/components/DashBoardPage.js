import React from "react"





export const DashBoardPage = () => (
    <div>
        Dashboard page Content
    </div>
)






export default DashBoardPage