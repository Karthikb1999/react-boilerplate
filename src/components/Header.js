import React from "react"
import { Link } from "react-router-dom"
import { connect } from "react-redux"
import { startLogout } from "../actions/auth"


export const Header = ({ startLogout }) => (
    <header className="header">
        <div className="container">
            <div className="header__container">
                <Link to="/dashboard" className="header__title">
                    <h1>Boilerplate</h1>
                </Link>
                <button
                    onClick={startLogout}
                    className="button button--header"
                >Logout</button>
            </div>
        </div>

    </header>
)

const mapDispatchtoProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchtoProps)(Header)