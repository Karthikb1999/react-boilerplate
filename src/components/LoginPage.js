import React from "react"
import { startLogin } from "../actions/auth"
import { connect } from "react-redux"


export const LoginPage = ({ startLogin }) => {
    return (
        <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__box__title">Boilerplate for app</h1>
                <p>-tag line</p>
                <button className="button" onClick={startLogin}>Login with Google</button>
            </div>
        </div>
    )
}

const mapDispatchtoProps = (dispatch) => ({
    startLogin: () => {
        dispatch(startLogin())
    }
})

export default connect(undefined, mapDispatchtoProps)(LoginPage)