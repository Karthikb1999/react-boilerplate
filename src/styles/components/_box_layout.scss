.box-layout{
    background-image: url('/images/bg.jpg');
    width: 100vw;
    height: 100vh;
    background-size: cover;
    display: flex;
    justify-content: center;
    align-items: center;
}

.box-layout__box{
    background-color: fade-out($color: white, $amount: 0.15);
    border-radius:3px;
    width:25rem;
    padding: $l-size $m-size;
    text-align: center;
}

.box-layout__box__title{
    margin:0;
    margin-bottom: $m-size;
    line-height: 1;
}