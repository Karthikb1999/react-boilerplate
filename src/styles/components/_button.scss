.button{
    border:0;
    background-color: $blue;
    color:white;
    padding: $s-size;
    font-weight: 300;
    font-size:$font-size-large;
    text-decoration: none;
    display: inline-block;
    line-height: 1;
}

.button--header{
    background:none ;
}

.button--secondary{
    @extend .button;
    background-color: #888888;
}