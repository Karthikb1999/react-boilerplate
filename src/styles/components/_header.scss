.header{
    background-color:$thick-blue ;
}


.header__title{
    text-decoration: none;
    color:white;
    h1{
        margin: 0;
    }
}

.header__container{
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: $s-size 0;
}

