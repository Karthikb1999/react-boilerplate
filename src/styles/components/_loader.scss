.loader{
    display: flex;
    height: 100vh;
    width:100vw;
    justify-content: center;
    align-items: center;
}

.loader__img{
    height: 6rem;
    width: 6rem;
}