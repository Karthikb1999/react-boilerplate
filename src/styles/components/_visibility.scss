.show-to-mobile{
    @media(min-width: $desktop-breakpoint){
        display: none;
    }
}

.show-to-desktop{
    @media(max-width: $desktop-breakpoint - 0.01rem){
        display: none;
    }
}