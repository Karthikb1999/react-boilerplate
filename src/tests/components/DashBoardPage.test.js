import { DashBoardPage } from "../../components/DashBoardPage"
import React from "react"
import { shallow } from "enzyme"

test("should render DashBoard Page", () => {
    const wrapper = shallow(<DashBoardPage />)
    expect(wrapper).toMatchSnapshot()
})