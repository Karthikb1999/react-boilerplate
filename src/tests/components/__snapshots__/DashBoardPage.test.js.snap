// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashBoard Page 1`] = `
<div>
  Dashboard page Content
</div>
`;
